contents/go/Go.md

```diff
@@ -0,0 +1,143 @@
+
+### Concurrency
+One of Go's special features is its support for concurrency. To this end, Go's standard library comes with two features that allow for easy and maintainable concurrency.
+
+#### Goroutines
+A goroutine is a lightweight thread that executes a function concurrently with its caller. A goroutine is launched by a `go` statement:
+
+```go
+func run() {
+  \\ does something
+}
+
+func main() {
+  go run()
+  executeOtherCommands()
+}
+```
+
+In the above example, using the `go` keyword launches a goroutine, which executes `run()` concurrently with `executeOtherCommands()`.
+
+Goroutines can also be started for anonymous functions:
+```go
+go func(msg string) {
+  fmt.Println(msg)
+}("a message")  // starts a goroutine that prints "a message"
+```
+
+A goroutine is not its own thread; instead, goroutines are dynamically multiplexed onto threads as required to keep them running. This makes them lightweight, so having a large number of goroutines is feasible. In practice, goroutines behave similarly to very cheap threads.
+
+#### Channels
+
+Channels are used to fulfill Go's philosophy on concurrent software: "don't communicate by sharing memory; share memory by communicating".
+
+Channels are typed conduits that allows goroutines to communicate with each other by sending and receiving messages. Before using a channel of a specific type, we must declare and `make` it:
+
+```go
+var c chan int
+c = make(chan int)
+
+//alternatively
+c := make(chan int)
+```
+
+Channels can transmit data of any type; thus, creating a channel that transmits channels (i.e. a `chan chan`) is theoretically possible and may even be [useful](http://tleyden.github.io/blog/2013/11/23/understanding-chan-chans-in-go/).
+
+Goroutines send and receive messages through a channel using the `<-` operator.
+
+```go
+ch <- v    // Send v to channel ch.
+v := <-ch  // Receive from ch, and
+           // assign value to v.
+```
+One useful way to think about sending and receiving data with the `<-` operator is that the data moves in the direction of the arrow.
+
+If you are interested in delving deeper into using Go's concurrency features extensively, Google developers have put out video presentations on Go's [basic](https://www.youtube.com/watch?v=f6kdp27TYZs) and [advanced concurrency patterns](https://www.youtube.com/watch?v=QDDwwePbDtw). This [code walkthrough](https://golang.org/doc/codewalk/sharemem/) provides an annotated example of how Go's memory-sharing principles can be applied in practice, .
+
+### Effective Go
+Formatting in Go is enforced by running `gofmt`, which will align your source code with the language-wide standard style of indentation and vertical alignment. Thus, given the following code:
+
+```go
+type T struct {
+    name string // name of the object
+    value int // its value
+}
+```
+Running `$ gofmt` in the same directory as the source file will line up comments and correct the source code's indentation:
```

Comment:
I think if you're specifying a path (like current directory) you need to use `$ go fmt` instead. Can you double-check?